src/actions/actions.ts

```diff
@@ -3537,20 +3537,55 @@ class ToggleCaseOperator extends BaseOperator {
 
   public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
     const range = new vscode.Range(start, new Position(end.line, end.character + 1));
-    const char = TextEditor.getText(range);
-
-    // Try lower-case
-    let toggled = char.toLocaleLowerCase();
-    if (toggled === char) {
-      // Try upper-case
-      toggled = char.toLocaleUpperCase();
+    const text = TextEditor.getText(range);
+
+    let newText = "";
+    for (var i = 0; i < text.length; i++) {
+      var char = text[i];
+      // Try lower-case
+      let toggled = char.toLocaleLowerCase();
```

Comment:
Instead of checking whether it was toggled, how do you guys feel about an ASCII hack :)

1) Check if the char is A-Za-z
2) XOR the char code with the ASCII for space (32). `String.fromCharCode(char.charCodeAt(0) ^ 32)`